src/js/dynload.ts

```diff
@@ -3,28 +3,53 @@
 declare var DEBUG: boolean;
 
 import { createLock } from "./lock";
-import { memoize } from "./pyodide-util";
 import { InternalPackageData } from "./load-package";
 import { LoadDynlibFS, ReadFileType } from "./types";
 
-// File System-like type which can be passed to
-// Module.loadDynamicLibrary or Module.loadWebAssemblyModule
+/**
+ * Recursively get all subdirectories of a directory
+ *
+ * @param dir The absolute path to the directory
+ * @returns A list of absolute paths to the subdirectories
+ * @private
+ */
+function getSubDirs(dir: string): string[] {
+  const dirs = Module.FS.readdir(dir);
+  const subDirs: string[] = [];
+
+  for (const d of dirs) {
+    if (d === "." || d === "..") {
+      continue;
+    }
+
+    const subdir = Module.PATH.join2(dir, d);
+    const lookup = Module.FS.lookupPath(subdir);
+    if (lookup.node === null) {
+      continue;
+    }
+
+    const mode = lookup.node.mode;
+    if (!Module.FS.isDir(mode)) {
+      continue;
+    }
+
+    subDirs.push(subdir);
+    subDirs.push(...getSubDirs(subdir));
+  }
+
+  return subDirs;
+}
```

Comment:
Actually this would be a great spot for a generator:
```suggestion
function* getSubDirs(dir: string): Iterator<string> {
  const dirs = Module.FS.readdir(dir);

  for (const d of dirs) {
    if (d === "." || d === "..") {
      continue;
    }

    const subdir = Module.PATH.join2(dir, d);
    const lookup = Module.FS.lookupPath(subdir);
    if (lookup.node === null) {
      continue;
    }

    const mode = lookup.node.mode;
    if (!Module.FS.isDir(mode)) {
      continue;
    }

    yield subdir;
    yield from getSubDirs(subdir);
  }
}
```

Reply:
Oh, right. I didn't think of using generators in JS. Thanks for the suggestion.